### PR TITLE
New abstract test class

### PR DESCRIPTION
**Pull request by Jacek-fstack**
Adds new abstract test to handle a new user data setup.

### CODE REVIEW

File: src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.formstack.qa.tests;
+
+import com.formstack.qa.model.SignupData;
+import com.formstack.qa.model.factories.SignupFactory;
+import com.formstack.qa.utils.Utils;
+import org.apache.log4j.Logger;
+import org.testng.annotations.BeforeMethod;
+
+
+public class AbstractNewUserTest extends AbstractTest {
```
Comment (kip9) [inline src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java]: IMO `AbstractAuthenticatedUserTest` sounds a bit more accurate - thoughts?

File: src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    final static Logger logger = Logger.getLogger(AbstractNewUserTest.class);
```
Comment (kip9) [inline src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java]: We should be able to have per class loggers if we define it this way:
```suggestion
    protected final Logger logger = Logger.getLogger(getClass());
```

With your solution, we either need to overload logger with every child class, or we'll get logs in the abstract class context, and that's not what we want.

File: src/test/java/com/formstack/qa/tests/BasicSubmissionsTest.java [hunk 3]
```diff
@@ -29,18 +25,6 @@
     private static final String FIRST_NAME = "Sample First Name";
     private static final String LAST_NAME = "Sample Last Name";
     private static final Address address = AnswersFactory.getDefaultAddress();
-    private Utils utils = new Utils();
-    private String link;
-    private String email;
-    private SignupData data;
-    private AnswersValidator answersValidator = new AnswersValidator();
```
Comment (kip9) [inline src/test/java/com/formstack/qa/tests/BasicSubmissionsTest.java]: Is that not needed anymore? It's not related to `AbstractNewUserTest` so wanted to double check

File: src/test/java/com/formstack/qa/tests/StandardAccountPCITest.java [hunk 4]
```diff
@@ -3,34 +3,25 @@
...
     private final String paymentPlan = "gold";
     private final String ccWarningMessage = "You must enable a credit card processing integration.";
 
     @BeforeMethod
     public void setup() {
-        email = utils.generateDefaultEmailAddress();
-        data = new SignupFactory().getDefaultSignupData(email);
-        link = utils.registerCustomer(data);
+        super.setup();
         billingData = new BillingFactory().getDefaultBillingData();
```
Comment (kip9) [inline src/test/java/com/formstack/qa/tests/StandardAccountPCITest.java]: Looks like there is an opportunity for one more Abstract base class, that extends the newly created `AbstractNewUserTest`, called `AbstractPayingAccountTest` ...
It's constructor should have that one extra property 

`private BillingData billingData;` 

and should invoke constructor with:
```
@BeforeMethod
    public void setup() {
        super.setup();
        billingData = new BillingFactory().getDefaultBillingData();
    }
```

This way you don't need `@BeforeMethod` in lots of tests at all I think.

**Comment by kip9** [comment]
Overall it's a great update - lots of lines of code removed/saved, and better path for refactoring in the future 👍

File: src/test/java/com/formstack/qa/tests/BasicSubmissionsTest.java (hunk 3 above)
Comment (Jacek-fstack) [reply src/test/java/com/formstack/qa/tests/BasicSubmissionsTest.java]: It's defined as an object, but we use it's static methods directly from the class, so this was never used.

File: src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java (hunk 1 above)
Comment (Jacek-fstack) [reply src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java]: Yep, that works as well.

File: src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java (hunk 2 above)
Comment (Jacek-fstack) [reply src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java]: So if we do it the way you're suggesting we can remove the logger from every child class?

File: src/test/java/com/formstack/qa/tests/StandardAccountPCITest.java (hunk 4 above)
Comment (Jacek-fstack) [reply src/test/java/com/formstack/qa/tests/StandardAccountPCITest.java]: Yep, that works, I can definitely do that.

File: src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java (hunk 2 above)
Comment (kip9) [reply src/test/java/com/formstack/qa/tests/AbstractNewUserTest.java]: Yes, that's the whole idea behind it. https://stackoverflow.com/a/6653577 has a bit more information.

**Comment by Jacek-fstack** [comment]
TestCase2 is already handled, will take care of the other two :)

**Pull request merged**